internal/server/warcserver/db.go

```diff
@@ -299,6 +227,93 @@ func (c DbCdxServer) sortedParallelSearch(api *CdxjServerApi, perCdxFunc PerCdxF
 	return count, err
 }
 
+func (c DbCdxServer) unsortedSerialSearch(api *CdxjServerApi, perCdxFunc PerCdxFunc) (int, error) {
+	keys := make([]string, len(api.Urls))
+	for i, url := range api.Urls {
+		key := parseKey(surt.UrlToSsurt(url), api.MatchType)
+		keys[i] = key
+	}
+
+	count := 0
+	err := c.Walk(func(txn *badger.Txn) error {
+		// initalize badger iterators
+		iterators := make([]*badger.Iterator, len(keys))
+		prefixes := make([][]byte, len(keys))
+		for i, key := range keys {
+			prefixes[i] = []byte(key)
+			opts := badger.DefaultIteratorOptions
+			opts.Prefix = prefixes[i]
+
+			iterators[i] = txn.NewIterator(opts)
```

Comment:
Add `defer iterator.Close()` to ensure it is closed if error is returned. The method is idempotent so the close call below is no problem.